core/storage/question/gae_models_test.py

```diff
@@ -199,3 +199,42 @@ def test_get_models_by_question_id(self):
                 'question_id3')
         )
         self.assertEqual(len(question_skill_links), 0)
+
+    def test_get_question_skill_links_with_constant_number_per_skill(self):
```

Comment:
In general, the tests need to be more robust. Consider cases like "not enough models in the datastore", etc. Try to think of other cases which could break things, and test them too.

Better to handle these now than find them out later in production.

Reply:
It would be handled automatically I think. For example, if we want three questions but there are only two questions in datastore, only those two questions will be fetched, so it wouldn't break things. And actually, this case is tested here, considering I was asking for four questions but there are only three questions exist.

Reply:
Ah OK. I don't buy the first argument, because just because things don't break now, doesn't mean they won't break in the future -- tests are supposed to handle future regressions. But I do buy the second, I misread it and thought you were requesting three -- so you're right, that case is tested. Sorry!

It would however be a good idea to add a comment or make it a separate function that explicitly says something about testing for queries that retrieve more questions than what is available. See [these guidelines](https://github.com/oppia/oppia/wiki/Writing-backend-tests#guidelines-for-writing-good-tests) for writing good backend tests.

Reply:
Ok got it. I will add a comment!

Reply:
Done.